Feature/aws policy checker (#34)
devonfw/hangar: documentation/aws/setup-aws-account-iam-for-eks.asciidoc

* An AWS account with IAM full access permission.

> 

Reviewer: ```suggestion

=== Alternative

```

In case you do not have an account or permission to create new IAM users, request it to your AWS administrator asking for the following policies being attached. Then go to <<check-iam-user-permissions>>.

> === Required managed policies

Reviewer: ```suggestion
==== Required managed policies
```

```
AmazonEC2FullAccess
IAMReadOnlyAccess
AmazonEKSServicePolicy
AmazonS3FullAccess
AmazonEC2ContainerRegistryFullAccess
```

> === Required custom policies

Reviewer: ```suggestion
==== Required custom policies
```

> In case you do not have an account or permission to create new IAM users, request it to your AWS administrator asking for the following policies attached. Then go to <<configure-cli>>.

Reviewer: ```suggestion
In case you do not have an account or permission to create new IAM users, request it to your AWS administrator asking for the following policies being attached. Then go to <<check-iam-user-permissions>>.

=== Required managed policies
```

```
AmazonEC2FullAccess
IAMReadOnlyAccess
AmazonEKSServicePolicy
AmazonS3FullAccess
AmazonEC2ContainerRegistryFullAccess

> eks:*

Reviewer: ```suggestion
```

```

> 

Reviewer: ```suggestion

=== Required custom policies

Find them on `/scripts/accounts/aws/eks-custom-policies.json`.

```

=== Usage
```

> create-user.sh -u <username> -g <group> [-p <policies...>] [-f <policies file path>] [-c "<custom policies file>"] [-a "<AWS access key>"] [-s "<AWS secret key>"] [-r <region>] 

Reviewer: ```suggestion
create-user.sh \
  -u <username> \
  -g <group> \
  [-p <policies...>] \
  [-f <policies file path>] \
  [-c <custom policies file path>] \
  [-a <AWS access key>] \
  [-s <AWS secret key>] \
  [-r <region>] 
```

-g      [Required] Group name for the group to be created or used
-p      [Optional] Policies to be attached to the group, splitted by comma
-f      [Optional] Path to a file containing the policies to be attached to the group

> -c     [Optional] Path to a json file containing the custom policies to be attached to the group.

Reviewer: ```suggestion
-c      [Optional] Path to a json file containing the custom policies to be attached to the group.
```

-a      [Optional] AWS administrator access key
-s      [Optional] AWS administrator secret key
-r      [Optional] AWS region
```

=== Example
```

> ./create-user.sh -u "Bob" -g "DevOps" -f "./eks-managed-policies.txt" -c "eks-custom-policies.json" -a "myAccessKey" -s "mySecretKey" -r "eu-west-1" 

Reviewer: ```suggestion
./create-user.sh -u Bob -g DevOps -f ./eks-managed-policies.txt -c ./eks-custom-policies.json -a "myAccessKey" -s "mySecretKey" -r eu-west-1
```

```
NOTE: If the "DevOps" group does not exist, it will be created.

> 

Reviewer: ```suggestion

NOTE: Required policies for EKS usage are located at `/scripts/accounts/aws/eks-managed-policies.txt` and `/scripts/accounts/aws/eks-custom-policies.json`

```

=== After execution
On success, the newly created user access data will be shown as output:
```
Access key ID: <accessKeyID>
Secret access key: <secretAccessKey>
```

IMPORTANT: It is mandatory to store the access key ID and the secret access key securely at this point, as *they will not be retrievable again*.

> NOTE: Required policies for EKS usage are located at `/scripts/accounts/aws/eks-managed-policies.txt` and `/scripts/accounts/aws/eks-custom-policies.json`
> 

Reviewer: ```suggestion
```

== Check user permissions

The script located at `/scripts/accounts/aws/verify-account-policies.sh` will check that policies were correctly attached to the user.

=== Usage
```

> verify-account-policies.sh -u <username> [-p <policies...>] [-f <policies file path>] [-c <custom policies file path>] [-a "<AWS access key>"] [-s "<AWS secret key>"] [-r <region>] 

Reviewer: ```suggestion
verify-account-policies.sh \
  -u <username> \
  [-p <policies...>] \
  [-f <policies file path>] \
  [-c <custom policies file path>] \
  [-a <AWS access key>] \
  [-s <AWS secret key>] \
  [-r <region>] 
```

```

=== Flags
```
-u      [Required] Username whose policies will be checked
-p      [Optional] Policies to be checked, splitted by comma
-f      [Optional] Path to a file containing the policies to be checked
-c      [Optional] Path to a file containing the custom policies to be checked
-a      [Optional] AWS administrator access key
-s      [Optional] AWS administrator secret key
-r      [Optional] AWS region
```
NOTE: At least one policies flag (`-p`, `-f` or `-c`) is required.

=== Example
```

> ./verify-account-policies.sh -u "Bob" -f "./eks-managed-policies.txt" -c "./eks-custom-policies.json"-a "myAccessKey" -s "mySecretKey" -r "eu-west-1" 

Reviewer: ```suggestion
./verify-account-policies.sh -u Bob -f ./eks-managed-policies.txt -c ./eks-custom-policies.json-a "myAccessKey" -s "mySecretKey" -r eu-west-1
```

-g      [Required] Group name for the group to be created or used
-p      [Optional] Policies to be attached to the group, splitted by comma
-f      [Optional] Path to a file containing the policies to be attached to the group
-c      [Optional] Path to a json file containing the custom policies to be attached to the group.
-a      [Optional] AWS administrator access key
-s      [Optional] AWS administrator secret key
-r      [Optional] AWS region
```

=== Example
```
./create-user.sh -u Bob -g DevOps -f ./eks-managed-policies.txt -c ./eks-custom-policies.json -a "myAccessKey" -s "mySecretKey" -r eu-west-1
```
NOTE: If the "DevOps" group does not exist, it will be created.

> NOTE: Required policies for EKS usage are located at `/scripts/accounts/aws/eks-managed-policies.txt` and `/scripts/accounts/aws/eks-custom-policies.json`

Reviewer: ```suggestion
NOTE: Required policies for using EKS are located at `/scripts/accounts/aws/eks-managed-policies.txt` and `/scripts/accounts/aws/eks-custom-policies.json`
```

=== After execution
On success, the newly created user access data will be shown as output:
```
Access key ID: <accessKeyID>
Secret access key: <secretAccessKey>
```

IMPORTANT: It is mandatory to store the access key ID and the secret access key securely at this point, as *they will not be retrievable again*.

> == Check user permissions

Reviewer: ```suggestion
== Check IAM user permissions
```

> The script located at `/scripts/accounts/aws/verify-account-policies.sh` will check that policies were correctly attached to the user.

Reviewer: ```suggestion
The script located at `/scripts/accounts/aws/verify-account-policies.sh` will check that the necessary policies were attached to the IAM user.
```

=== Usage
```
verify-account-policies.sh \
  -u <username> \
  [-p <policies...>] \
  [-f <policies file path>] \
  [-c <custom policies file path>] \
  [-a <AWS access key>] \
  [-s <AWS secret key>] \
  [-r <region>] 
```

=== Flags
```
-u      [Required] Username whose policies will be checked
-p      [Optional] Policies to be checked, splitted by comma
-f      [Optional] Path to a file containing the policies to be checked
-c      [Optional] Path to a file containing the custom policies to be checked
-a      [Optional] AWS administrator access key
-s      [Optional] AWS administrator secret key
-r      [Optional] AWS region
```
NOTE: At least one policies flag (`-p`, `-f` or `-c`) is required.

=== Example
```
./verify-account-policies.sh -u Bob -f ./eks-managed-policies.txt -c ./eks-custom-policies.json-a "myAccessKey" -s "mySecretKey" -r eu-west-1
```
After execution, provided policies will be shown preceded by an `OK` or `FAILED` depending on the attachment status.

> NOTE: Required policies for EKS usage are located at `/scripts/accounts/aws/eks-managed-policies.txt` and `/scripts/accounts/aws/eks-custom-policies.json`

Reviewer: ```suggestion
NOTE: Required policies for using EKS are located at `/scripts/accounts/aws/eks-managed-policies.txt` and `/scripts/accounts/aws/eks-custom-policies.json`
```

== Check user permissions

The script located at `/scripts/accounts/aws/verify-account-policies.sh` will check that policies were correctly attached to the user.

=== Usage
```
verify-account-policies.sh -u <username> [-p <policies...>] [-f <policies file path>] [-c <custom policies file path>] [-a "<AWS access key>"] [-s "<AWS secret key>"] [-r <region>] 
```

=== Flags
```
-u      [Required] Username whose policies will be checked
-p      [Optional] Policies to be checked, splitted by comma
-f      [Optional] Path to a file containing the policies to be checked
-c      [Optional] Path to a file containing the custom policies to be checked
-a      [Optional] AWS administrator access key
-s      [Optional] AWS administrator secret key
-r      [Optional] AWS region
```
NOTE: At least one policies flag (`-p`, `-f` or `-c`) is required.

=== Example
```
./verify-account-policies.sh -u "Bob" -f "./eks-managed-policies.txt" -c "./eks-custom-policies.json"-a "myAccessKey" -s "mySecretKey" -r "eu-west-1" 
```
After execution, provided policies will be shown preceded by an `OK` or `FAILED` depending on the attachment status.

> NOTE: Required policies for EKS usage are located at `/scripts/accounts/aws/eks-managed-policies.txt` and `/scripts/accounts/aws/eks-custom-policies.json`

Reviewer: ```suggestion
NOTE: Required policies for using EKS are located at `/scripts/accounts/aws/eks-managed-policies.txt` and `/scripts/accounts/aws/eks-custom-policies.json`
```

Reviewer: It would be good to include this note also in the first section example, below the existing note.
